### PR TITLE
Generate cylc8 syntax

### PR DESCRIPTION
**Pull request by jim-p-w**
### Description
The python code which generates the cylc configuration files was generating cylc 7 compliant configurations.
Cylc 7 is deprecated, and cylc 8 is being used on derecho. So the files which generate the cylc configurations need to be updated. 
### Issue closed
[Issue 267](https://github.com/NCAR/MPAS-Workflow/issues/267)
Closes #(if applicable)

#### Tier 1:
 - [x] 3dvar_OIE120km_WarmStart
 - [x] 3denvar_OIE120km_IAU_WarmStart
 - [x] 3dvar_OIE120km_ColdStart
 - [x] 3dvar_O30kmIE60km_ColdStart
 - [x] 3denvar_O30kmIE60km_WarmStart
 - [x]  4denvar_OIE120km_WarmStart
 - [x] eda_OIE120km_WarmStart
 - [x] getkf_OIE120km_WarmStart
 - [x] ForecastFromGFSAnalysesMPT

#### Tier 2 (optional):
 - [ ] GenerateGFSAnalyses
 - [ ] GenerateObs
 
### Issues to be addressed
The changes will eliminate the following warning messages, which are generated when running `cylc validate`

1. WARNING - deprecated graph items were automatically upgraded in "workflow definition":
     * (8.0.0) [scheduling][dependencies][X]graph -> [scheduling][graph]X - for X in:
           T00
2. WARNING -  * (8.0.0) [runtime][BATCH][job]execution time limit -> [runtime][BATCH]execution time limit - value unchanged
3. WARNING -  * (8.0.0) [runtime][LinkExternalAnalysis-120km-0hr][job]execution retry delays -> [runtime][LinkExternalAnalysis-120km-0hr]execution retry delays - value unchanged
4. WARNING -  * (8.0.0) [scheduling]max active cycle points -> [scheduling]runahead limit - "4" -> "P3"
5. WARNING - The event handler template variable "%(suite)s" is deprecated - use "%(workflow)s" instead.
6. WARNING -  * (8.0.0) [runtime][root][events]submission timeout handler -> [runtime][root][events]submission timeout handlers - value unchanged
7. WARNING -  * (8.0.0) [cylc] -> [scheduler] - value unchanged
8. WARNING -  * (8.0.0) [visualization] - DELETED (OBSOLETE)
9. WARNING - deprecated settings found (please replace with [runtime][BATCH]platform):
    [runtime][BATCH][job]batch system = pbs

### CODE REVIEW

File: .gitignore [hunk 1]
```diff
@@ -8,3 +8,6 @@ __pycache__
 flow.cylc
 
 config/jedi/ObsPlugs/retrieveObsErrors/*.yaml
+
+# ctags symbol file
```
Comment (jim-p-w) [inline .gitignore:12]: This is the tags database file, used to jump directly to a symbol definition when using vi. I'm happy to demonstrate if you are interested.

File: initialize/applications/DA.py [hunk 2]
```diff
@@ -166,8 +166,7 @@ def export(self, previousForecast:str, ef:ExtendedForecast):
     ########################
     # open graph
     self._dependencies += ['''
-    [[['''+self.workflow['AnalysisTimes']+''']]]
-      graph = """''']
+    '''+self.workflow['AnalysisTimes']+''' = """''']
```
Comment (jim-p-w) [inline initialize/applications/DA.py:169]: The old syntax was
```
 [[[+PT6H/PT6H]]]
  graph = """
  ...
```
The new syntax is:
``` 
+PT6H/PT6H = """
...
```

File: initialize/applications/EnKF.py [hunk 3]
```diff
@@ -204,16 +204,14 @@ def __init__(self,
   [[InitEnKF_0]]
     inherit = '''+self.tf.init+''', SingleBatch
     script = $origin/bin/PrepJEDI.csh '''+initArgs+'''
-    [[[job]]]
-      execution time limit = PT10M
-      execution retry delays = '''+solverjob['retry']+'''
+    execution time limit = PT10M
+    execution retry delays = '''+solverjob['retry']+'''
```
Comment (jim-p-w) [inline initialize/applications/EnKF.py:208]: The old syntax was:
```
  [[Clean]]
    [[[job]]]
      execution time limit = PT5M
```
The new syntax is:
```
  [[Clean]]
    execution time limit = PT5M
```

File: initialize/config/Task.py [hunk 4]
```diff
@@ -21,12 +21,12 @@ def job(self):
...
-    [[[job]]]
-      batch system = '''+self.batchSystem+'''
-      execution time limit = PT'''+str(int(seconds))+'S'
+    execution time limit = PT'''+str(int(seconds))+'S'
     if retry is not None:
       text += '''
-      execution retry delays = '''+str(retry)
+    execution retry delays = '''+str(retry)
+    text += '''
+    platform = '''+self.batchSystem+'_cluster'
```
Comment (jim-p-w) [inline initialize/config/Task.py:29]: The old syntax was:
```
    [[[job]]]
      batch system = pbs
```
The new syntax is:
```
platform = pbs_cluster
```

File: initialize/suites/SuiteBase.py [hunk 5]
```diff
@@ -171,22 +170,16 @@ def __export(self):
...
 
 [runtime]
-'''+''.join(self._tasks)+'''
-
-[visualization]
-  initial cycle point = '''+self.c['workflow']['restart cycle point']+'''
-  final cycle point   = '''+self.c['workflow']['final cycle point']+'''
-  number of cycle points = 200
-  default node attributes = "style=filled", "fillcolor=grey"''']
+'''+''.join(self._tasks)]
```
Comment (jim-p-w) [inline initialize/suites/SuiteBase.py:182]: The visualization section is obsolete and no longer used.

File: initialize/framework/Workflow.py [hunk 6]
```diff
@@ -23,7 +23,7 @@ class Workflow(Component):
     'CyclingWindowHR': [6, int, [6]],
 
     # maximum consecutive cycle points to be active at any time
-    'max active cycle points': [4, int],
+    'max active cycle points': [3, int],
```
Comment (liujake) [inline initialize/framework/Workflow.py]: why this needs to change from 4 to 3?

File: initialize/suites/SuiteBase.py (hunk 5 above)
Comment (liujake) [reply initialize/suites/SuiteBase.py:182]: The 'visualization' section was for Cylc GUI? i.e., when using 'cylc gscan'?

File: initialize/framework/Workflow.py (hunk 6 above)
Comment (jim-p-w) [reply initialize/framework/Workflow.py]: I believe this change is incorrect!
It should read:
`runahead limit = P3  # max 4`
I will investigate when derecho is back online.
According to the cylc docs:
 `The integer format Pn is an interval that spans n+1 consecutive cycle points`

Comment (ibanos90) [reply initialize/framework/Workflow.py]: @jim-p-w, it seems like indeed in cylc v8.2 the number of maximum active cycle points is `n+1`, then we need to specify `n-1` if we want to keep our same configurations. However, in order to not make any further changes in our scenarios or other yaml files, we can subtract 1 when specifying this in `SuiteBase.py`, e.g: 
```runahead limit = P'''+str(self.c['workflow']['max active cycle points']-1)+'''```

Comment (jim-p-w) [reply initialize/framework/Workflow.py]: @ibanos90 Great idea. I will make that change, and undo the change (4 => 3) in this conversation.

Comment (ibanos90) [reply initialize/framework/Workflow.py]: Great, thanks!

File: initialize/suites/SuiteBase.py (hunk 5 above)
Comment (jim-p-w) [reply initialize/suites/SuiteBase.py:182]: @liujake The visualization section allowed stylizing the output of the `cylc graph` command.
See https://cylc.github.io/cylc-doc/7.8.8/html/tutorial.html#suite-visualization

File: initialize/framework/Workflow.py (hunk 6 above)
Comment (jim-p-w) [reply initialize/framework/Workflow.py]: The changes are pushed. The test suite still passes.

**Comment by jim-p-w** [comment]
@liujake Do we want these changes (after they are approved) to get merged into develop, and then tag v2.1.0?

**Review by liujake — APPROVED** [review]

**Comment by liujake** [comment]
Ok with me to merge. But I think we'd better to have @RGNystrom 's fix about hourly output before making a 2.1.0 branch/tag. You could experiment with the develop branch for now, which works with 2.1 bundle code.

**Review by ibanos90 — APPROVED** [review]
The changes look good to me and the tests are all completing successfully. Thanks @jim-p-w!

**Pull request merged**